editoast/src/core/simulation.rs

```diff
@@ -475,6 +478,21 @@ pub struct SimulationRequest {
     pub electrical_profile_set_id: Option<i64>,
 }
 
+impl SimulationRequest {
+    // Compute hash input of a simulation
+    pub fn compute_train_simulation_hash_with_versioning(
+        &self,
+        infra_id: i64,
+        infra_version: &String,
+    ) -> String {
+        let osrd_version = get_app_version().unwrap_or_default();
+        let mut hasher = DefaultHasher::new();
+        self.hash(&mut hasher);
+        let hash_simulation_input = hasher.finish();
+        format!("simulation_{osrd_version}.{infra_id}.{infra_version}.{hash_simulation_input}")
+    }
+}
+
```

Comment:
This looks misplaced to me. `mod core` is just an HTTP (ish) client, the cache key computation is a view logic. The `core` client doesn't even know about redis....